Week3/api.js

```diff
@@ -0,0 +1,218 @@
+const mysql = require('mysql');
+const util = require('util');
+const { handelError } = require('./dbUtils');
+
+const connection = mysql.createConnection({
+  host: 'localhost',
+  user: 'hyfuser',
+  password: 'hyfpassword',
+  database: 'todos_app',
+});
+
+const execQuery = util.promisify(connection.query.bind(connection));
+connection.connect(handelError('connecting...'));
+
+class Api {
+  async getUsersHandler(req, res) {
+    try {
+      const users = await execQuery(`SELECT * FROM todos_app.users`);
+      if (!users.length) return res.status(404).send(`You need to insert some users!`);
+      res.send(users);
+    } catch (err) {
+      console.log(err.message);
+    }
+  }
+
+  async getUserHandler(req, res) {
+    const userId = req.params.id;
+    try {
+      const user = await execQuery(`SELECT * FROM todos_app.users WHERE User_id = ?`, [userId]);
+      if (!user.length) return res.status(404).send('The user with the given Id was not Found!');
+      res.send(user);
+    } catch (err) {
+      console.log(err);
+    }
+  }
+
+  async createUserHandler(req, res) {
+    try {
+      const { Name, User_id, Email } = req.body;
+      const create = `INSERT INTO todos_app.users (User_id, Name, Email) VALUES (?, ?, ?)`;
```

Comment:
User_id could have been auto increment, it's hard to ask people to provide their own primary key.

Reply:
Thank you very much to review my homework and give feedback! 
I don't understand but I have auto increment: 

``CREATE TABLE `users` (
  `user_id` int(11) NOT NULL AUTO_INCREMENT,
  `Name` varchar(45) DEFAULT NULL,
  `Email` varchar(60) DEFAULT NULL,
  PRIMARY KEY (`user_id`)
)``
I don't understand here what to change? Can you please give more detail?

Reply:
Yes you have it in the db but you don't use it, if you want to use it, either you put null at User_id or you provide the column name in your insert and you don't give id column. then sql will fill the field by itself.

Reply:
OK, I got the point, 
 const { Name, Email } = req.body;
 const create = `INSERT INTO todos_app.users (Name, Email) VALUES (?, ?)`;
 await execQuery(create, [Name, Email]);